orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationEmbeddingRequest.java

```diff
@@ -0,0 +1,176 @@
+package com.sap.ai.sdk.orchestration;
+
+import static com.sap.ai.sdk.orchestration.OrchestrationEmbeddingRequest.TokenType.DOCUMENT;
+import static com.sap.ai.sdk.orchestration.OrchestrationEmbeddingRequest.TokenType.QUERY;
+import static com.sap.ai.sdk.orchestration.OrchestrationEmbeddingRequest.TokenType.TEXT;
+import static lombok.AccessLevel.PRIVATE;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.Lists;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsInput;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsInputText;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsModelConfig;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsModuleConfigs;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsOrchestrationConfig;
+import com.sap.ai.sdk.orchestration.model.EmbeddingsPostRequest;
+import com.sap.ai.sdk.orchestration.model.MaskingModuleConfigProviders;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.With;
+import lombok.experimental.Tolerate;
+
+/**
+ * Represents a request for generating embeddings through the SAP AI Core Orchestration service.
+ *
+ * @since 1.11.0
+ */
+@Beta
+@Value
+@AllArgsConstructor(access = PRIVATE)
+public class OrchestrationEmbeddingRequest {
+
+  /** The embedding model to use for generating vector representations. */
+  @Nonnull OrchestrationEmbeddingModel model;
+
+  /** The list of text inputs to be converted into embeddings. */
+  @Nonnull List<String> tokens;
+
+  /** Optional masking providers for data privacy and security. */
+  @With(value = PRIVATE)
+  @Nullable
+  List<MaskingProvider> masking;
+
+  /** Optional token type classification to optimize embedding generation. */
+  @With(value = PRIVATE)
+  @Nullable
+  TokenType tokenType;
+
+  /**
+   * Create an embedding request using fluent API starting with model selection.
+   *
+   * <pre>{@code
+   * OrchestrationEmbeddingRequest.forModel(myModel).forInputs("text to embed");
+   * }</pre>
+   *
+   * @param model the embedding model to use
+   * @return a step for specifying inputs
+   */
+  @Nonnull
+  public static InputStep forModel(@Nonnull final OrchestrationEmbeddingModel model) {
+    return tokens -> new OrchestrationEmbeddingRequest(model, List.copyOf(tokens), null, null);
+  }
+
+  /** Builder step for specifying text inputs to embed. */
+  @FunctionalInterface
+  public interface InputStep {
```

Comment:
(Minor)

I'm not sure whether we used "Step" as "Builder" substitute already somewhere in the project. If not, please reconsider.